packages/@aws-cdk/aws-route53/test/vpc-endpoint-service-domain-name.test.ts

```diff
@@ -237,12 +238,6 @@ test('create domain name resource', () => {
 test('throws if creating multiple domains for a single service', () => {
   // GIVEN
 
-  new VpcEndpointServiceDomainName(stack, 'EndpointDomain', {
-    endpointService: vpces,
-    domainName: 'my-stuff-1.aws-cdk.dev',
-    publicHostedZone: zone,
-  });
```

Comment:
This seems kind of critical to this test... the test is verifying it throws when two domains are created for one service. Is the other domain being created elsewhere that I'm not seeing? Also -- it's a bit above and beyond -- adding the message to the `.toThrow()` below would go a long way to verifying we're actually seeing the right error.

Reply:
So basically what's going on here is the cache that is maintained by this construct is via a private `static` member. Since it is `static`, the cache is not clear across the tests. Now, we are creating one more such construct in the first test which is causing the error to occur. Hence we do not need two constructs here.

I will add the message here for the assertion as well

Reply:
Nevermind...I figured out a better way.